Lib/test/libregrtest/main.py

```diff
@@ -432,14 +433,26 @@ def display_header(self):
         print("== encodings: locale=%s, FS=%s"
               % (locale.getpreferredencoding(False),
                  sys.getfilesystemencoding()))
+        rl_version_cmd = """if 1:
         try:
             import readline
         except ImportError:
             readline = None
         if readline is not None:
-            print(f"== readline version: {readline._READLINE_VERSION:#06x}")
+            print(f"{readline._READLINE_VERSION:#06x}")
             is_editline = readline.__doc__ and "libedit" in readline.__doc__
-            print("== readline implementation:", "editline" if is_editline else "GNU readline")
+            print("editline" if is_editline else "GNU readline")
+        """
```

Comment:
More like:

````python
    def readline_info():
        script = """
import json
import readline
is_editline = readline.__doc__ and "libedit" in readline.__doc__
info = {"version": readline._READLINE_VERSION,
           "implementation": "editline" if is_editline else "GNU readline"}
print(json.dumps(info))
"""
        c = subprocess.run([sys.executable, "-c", script],
                                        stdout=subprocess.PIPE,
                                        stderr=subprocess.PIPE)
        if c.returncode == 0:
            return json.loads(c.stdout.decode())
```

And use readline_info() in display_header(), or in any other context as needed.